liquid-compiler/src/registry.rs

```diff
@@ -21,8 +21,8 @@ impl<P> PluginRegistry<P> {
     /// Generally this is used when setting up the program.
     ///
     /// Returns whether this overrode an existing plugin.
-    pub fn register(&mut self, name: &'static str, plugin: P) -> bool {
-        let old = self.plugins.insert(name, plugin);
+    pub fn register(&mut self, name: &str, plugin: P) -> bool {
+        let old = self.plugins.insert(name.to_string(), plugin);
```

Comment:
So using `&'static str` was an optimization that saw huge gains when implemented but things have changed since then.

Could you run and post the results of `cargo +nightly bench` in both master and your branch so we can see if this has any negative impact?

If there is, I have an idea for it.  For my changes, I'm thinking of creating a new string type that is smarter than `Cow<'s, str>`.  Something like

```rust
enum RenameMeString<'s> {
    Owned(String),
    Static(&'static str),
    Borrowed(&'s str),
}
```

This will let us preserve our `'static` optimizations will allowing borrowed/owned to be mixed in.

Reply:
I'll do the bench, didn't know of that command :D awesome!

Reply:
master: https://gist.github.com/Geobert/897517dd4680e9cc64dedf5ca22b1000
relaxed: https://gist.github.com/Geobert/e1f0207f221b1d0be640f4a074b76bde

how do we read the numbers? What the one in parenthesis means?

Reply:
The framework runs each benchmark a bunch of times and returns the average.  The parens represent how much spread there is above and below the average.

One thing I do when comparing numbers is only care when one of the numbers is outside of the range of the other.

Overall  this looks fine.  Looks like I took out all those extra clones that required the original optimization.